README.md

```diff
@@ -211,6 +211,128 @@ If you're using VSCode and running the MCP server in SSE mode (which is the defa
+
+1. **Configure OpenTelemetry infrastructure** (one-time setup):
+
+```go
+import (
+    "go.opentelemetry.io/otel"
+    "go.opentelemetry.io/otel/exporters/jaeger"  // or your preferred exporter
+    "go.opentelemetry.io/otel/sdk/trace"
+)
+
+// Set up tracer provider
+exporter, _ := jaeger.New(jaeger.WithCollectorEndpoint())
+tp := trace.NewTracerProvider(
+    trace.WithBatcher(exporter),
+    trace.WithResource(resource.NewWithAttributes(
+        semconv.ServiceName("my-app"),
+    )),
+)
+otel.SetTracerProvider(tp)
+```
+
+2. **Configure MCP server context**:
+
+```go
+grafanaConfig := mcpgrafana.GrafanaConfig{
+    URL:    "http://localhost:3000",
+    APIKey: "your-api-key",
+    EnableTracing: false,        // Default: no tracing overhead
+    LogTraceArguments: false,    // Default: safe (no PII logging)
+}
+
+srv.SetContextFunc(mcpgrafana.ComposedStdioContextFunc(grafanaConfig))
+```
+
+## **Enabling Complete Tracing**
+
+To get **full observability** (tool spans + HTTP request spans):
+
+```go
+config := mcpgrafana.GrafanaConfig{
+    EnableTracing: true,  // Enables both tool and HTTP tracing
+}
+```
+
+**Result:** Complete trace hierarchy with perfect correlation:
+
+```
+mcp.tool.query_prometheus [trace: abc123]
+├── http.request [trace: abc123, span: def456]
+│   ├── method: POST
+│   ├── url: /api/datasources/proxy/uid/abc123/api/v1/query
+│   └── status: 404
+└── error: "datasource not found"
+```
+
+## **Enabling Argument Logging**
+
+**⚠️ Privacy Warning:** Tool arguments may contain PII (emails, names, API tokens). Only enable when safe:
+
+```go
+config := mcpgrafana.GrafanaConfig{
+    EnableTracing: true,        // Required for any tracing
+    LogTraceArguments: true,    // Enable argument logging (use cautiously)
+}
+```
+
+## **Configuration Examples**
+
+**Production (safe):**
+
+```go
+config := mcpgrafana.GrafanaConfig{
+    EnableTracing: true,         // Full observability
+    LogTraceArguments: false,    // Keep it safe (default)
+}
+```
+
+**Development/debugging:**
+
+```go
+config := mcpgrafana.GrafanaConfig{
+    EnableTracing: true,         // Full observability
+    LogTraceArguments: true,     // Debug with arguments (when safe)
+}
+```
+
+**No tracing (default):**
+```go
+config := mcpgrafana.GrafanaConfig{
+    EnableTracing: false,        // Minimal overhead (default)
+}
+```
+
+## **What You Get**
+
+**Automatic tracing for every tool call:**
+
+```
+Your Application
+├── your.business.logic
+├── mcp.tool.list_teams (150ms, status=OK)
+├── mcp.tool.query_prometheus (2.3s, status=ERROR)
+│   └── error: "datasource not found"
+└── mcp.tool.search_dashboards (45ms, status=OK)
+```
+
+**With argument logging enabled:**
+
+```
+mcp.tool.query_prometheus
+├── attributes: {
+│   ├── mcp.tool.name: "query_prometheus"
+│   ├── mcp.tool.description: "Query Prometheus using PromQL..."
+│   └── mcp.tool.arguments: "{\"expr\":\"up\",\"datasourceUID\":\"abc123\"}"
+│   }
+```
+
```

Comment:
This shouldn't be in the main README IMO since it only applies to people using the MCP server programmatically, which is surely a small minority of users.

Reply:
I think I agree, pulled it out